components/calendar/src/chinese.rs

```diff
@@ -137,28 +137,13 @@ impl Calendar for Chinese {
             ));
         };
 
-        if month > Self::months_for_every_year(year) {
-            return Err(CalendarError::UnknownMonthCode(
-                month_code.0,
-                self.debug_name(),
-            ));
-        }
-
-        let max_day = Self::month_days(year, month);
-        if day > max_day {
-            return Err(CalendarError::Overflow {
-                field: "day",
-                max: max_day as usize,
-            });
-        }
-
         if era.0 != tinystr!(16, "chinese") {
             return Err(CalendarError::UnknownEra(era.0, self.debug_name()));
         }
 
-        Ok(ArithmeticDate::new_unchecked(year, month, day))
-            .map(ChineseBasedDateInner)
-            .map(ChineseDateInner)
+        let cache = Inner::compute_cache(year);
```

Comment:
Suggestion: Just noticed that this function already computes most of the cache data up above inside `chinese_based_ordinal_lunar_month_from_code`. Please refactor so that we can get the data out and only compute the parts we need. (see if you can do this for Dangi, too)